ArmoryQt.py

```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
 ############################################
 
 if 1:
-   #setup splash screen
+   # 1) Start the bridge BEFORE anything else (singleton) so dialogs can query it
+   #    Start with a no-op callback; we'll wire readiness to the dialog instance
+   #    below to avoid overwriting global callbacks.
+   TheBDM.startBridge(getBridgeArgList(), lambda *args, **kwargs: None)
+
+   #    Show Setup Manager (modal and focused)
+   #    Bridge is already running; dialog must not try to start it
+   from qtdialogs.DlgSetupManager import DlgSetupManager
+
+   dlg = DlgSetupManager(parent=None, main=None)
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
```

Comment:
this whole block makes no sense:

1. why check if the bridge is already running? this is the entry point to the process, it 100% knows if the bridge `start` method has been called by now
2. you can pass a ready callback `TheBDM.startBridge`, instead you do some lambda nonsense and access the inner connection object to force a callback after the start.
3. By refusing to use the code as its designed, you are introducing 2 race conditions: if the bridge readies up before you set the callback, it will never. If the dialog isn't started before the callback fires, it will try to progress over an undefined state of DlgSetupManager.

This is bad, use the code as it is meant to, don't adlib the start sequence.